File: app/src/main/java/com/example/myapplication/auth/SplashActivity.java
package com.example.myapplication.auth;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import androidx.appcompat.app.AppCompatActivity;
import com.example.myapplication.MainActivity;
import com.example.myapplication.R;

public class SplashActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        new Handler(Looper.getMainLooper()).postDelayed(() -> {
            SharedPreferences prefs = getSharedPreferences("circlo_prefs", MODE_PRIVATE);
            boolean isLoggedIn = prefs.getInt("user_id", 0) != 0;
            if (isLoggedIn) {
                startActivity(new Intent(this, MainActivity.class));
            } else {
                startActivity(new Intent(this, SignUpActivity.class));
            }
            finish();
        }, 2000);
    }
}

File: app/src/main/java/com/example/myapplication/auth/SplashActivity.kt
package com.example.myapplication.auth

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity
import com.example.myapplication.MainActivity
import com.example.myapplication.R
import com.example.myapplication.network.SupabaseClientProvider
import io.github.jan.supabase.auth.auth

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        Handler(Looper.getMainLooper()).postDelayed({
            val session = SupabaseClientProvider.client.auth.currentSessionOrNull()
            if (session != null) {
                startActivity(Intent(this, MainActivity::class.java))
            } else {
                startActivity(Intent(this, LoginOrSignUpActivity::class.java))
            }
            finish()
        }, 2000)
    }
}